include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -2639,13 +2648,13 @@ FEPointEvaluation<n_components_, dim, spacedim, Number>::evaluate_fast(
                v < stride && (stride == 1 || offset < this->n_q_points_scalar);
                ++v, ++offset)
             {
-              gradient_type unit_gradient;
+              unit_gradient_type unit_gradient;
               ETT::set_gradient(gradient, v, unit_gradient);
               this->gradients[offset] =
                 this->cell_type <=
                     internal::MatrixFreeFunctions::GeometryType::cartesian ?
-                  apply_diagonal_transformation(this->inverse_jacobian_ptr[0],
-                                                unit_gradient) :
+                  apply_diagonal_transformation(
+                    this->inverse_jacobian_ptr[0].transpose(), unit_gradient) :
```

Comment:
Are you sure about this? Why is the transpose necessary here?

Reply:
We need to transpose here to have consistent dimensions for the function (dim/spacedim), even though we only ever access the diagonal. In terms of correctness, it should not matter, but I also think it is more readable because also the `else` case in this check uses the transposed Jacobian. My assumption was that the transposed matrix is never built in release mode, because the entries are unused.

Reply:
Then we also need it for the `do_evaluate_in_face` function. Can you add a test for faces?

I am not sure about the transposition. I thought about an `apply_transposed_transformation` function in the past to make this explicit. We should check this in the machine code at some point, but I think it is not necessary now.